File: typescipt/existing/src/language/expressions/functions/daysFunction.ts


namespace Lexy.Compiler.Language.Expressions.Functions;

public class DaysFunction : EndStartDateFunction
{
   public const string Name = "DAYS";

   protected override string FunctionName => Name;

   private DaysFunction(Expression endDateExpression, Expression startDateExpression, SourceReference reference)
     : base(endDateExpression, startDateExpression, reference)
   {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression endDateExpression,
     Expression startDateExpression)
   {
     return new DaysFunction(endDateExpression, startDateExpression, reference);
   }
}

File: src/Lexy.Compiler/Language/Expressions/Functions/DaysFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions;

public class DaysFunction : EndStartDateFunction
{
    public const string Name = "DAYS";

    private DaysFunction(Expression endDateExpression, Expression startDateExpression, ExpressionSource source)
        : base(Name, endDateExpression, startDateExpression, source)
    {
    }

    public static FunctionCallExpression Create(ExpressionSource source, Expression endDateExpression,
        Expression startDateExpression)
    {
        return new DaysFunction(endDateExpression, startDateExpression, source);
    }
}